haystack/components/generators/chat/openai.py

```diff
@@ -280,9 +312,11 @@ def run(
             tools=tools,
             tools_strict=tools_strict,
         )
-        chat_completion: Union[Stream[ChatCompletionChunk], ChatCompletion] = self.client.chat.completions.create(
-            **api_args
-        )
+        openai_endpoint = api_args.pop("openai_endpoint")
+        if openai_endpoint == "parse":
+            chat_completion = self.client.chat.completions.parse(**api_args)
+        else:
+            chat_completion = self.client.chat.completions.create(**api_args)
```

Comment:
```suggestion
        openai_endpoint_method = getattr(self.client.chat.completions, openai_endpoint)
        chat_completion = openai_endpoint_method(**api_args)
```

just an idea to (maybe) simplify the code. Feel free to discard it if it's unclear. WDYT?